packages/unuse-angular/src/index.spec.ts

```diff
@@ -0,0 +1,190 @@
+// @vitest-environment happy-dom
+
+import {
+  computed,
+  NgModule,
+  provideZonelessChangeDetection,
+  signal,
+} from '@angular/core';
+import { getTestBed } from '@angular/core/testing';
+import {
+  BrowserTestingModule,
+  platformBrowserTesting,
+} from '@angular/platform-browser/testing';
+import { beforeAll, describe, expect, it, vi } from 'vitest';
+import {
+  isUnComputed,
+  isUnRef,
+  isUnSignal,
+  overrideIsUnRefFn,
+  overrideToUnSignalFn,
+  overrideTryOnScopeDisposeFn,
+  overrideUnResolveFn,
+  toUnSignal,
+  tryOnScopeDispose,
+  unComputed,
+  unResolve,
+  unSignal,
+} from '.';
+
+@NgModule({ providers: [provideZonelessChangeDetection()] })
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+class ZonelessTestModule {}
+
+describe('unuse-angular', () => {
+  it('should export correct global variable', () => {
+    expect(globalThis.__UNUSE_FRAMEWORK__).toBe('angular');
+  });
+
+  beforeAll(() => {
+    getTestBed().initTestEnvironment(
+      [BrowserTestingModule, ZonelessTestModule],
+      platformBrowserTesting()
+    );
+  });
+
+  describe('toUnSignal', () => {
+    it('should export toUnSignal function', () => {
+      expect(toUnSignal).toBeTypeOf('function');
+    });
+
+    it('should convert an Angular Signal to an UnSignal', () => {
+      const angularSignal = signal(42);
+      getTestBed().runInInjectionContext(() => {
+        const mySignal = toUnSignal(angularSignal);
+        expect(mySignal.get()).toBe(42);
+
+        mySignal.set(100);
+      });
+
+      expect(angularSignal()).toBe(100);
+    });
+
+    it('should convert a non-reactive value to an UnSignal', () => {
+      const mySignal = toUnSignal(42);
+      expect(mySignal.get()).toBe(42);
+
+      mySignal.set(100);
+      expect(mySignal.get()).toBe(100);
+    });
+  });
+
+  describe('unResolve', () => {
+    it('should export unResolve function', () => {
+      expect(unResolve).toBeTypeOf('function');
+    });
+
+    it('should resolve an UnSignal to an Angular WritableSignal', () => {
+      const mySignal = unSignal(42);
+      const resolved = unResolve(mySignal);
+
+      expect(resolved()).toBe(42);
+      resolved.set(100);
+      expect(mySignal.get()).toBe(100);
+    });
+
+    it('should resolve an UnSignal to an Angular Signal when readonly is true', () => {
+      const mySignal = unSignal(42);
+      const resolved = unResolve(mySignal, { readonly: true });
+
+      expect(resolved()).toBe(42);
+    });
+
+    it('should resolve an UnSignal to an UnSignal when framework is null', () => {
+      const mySignal = unSignal(42);
+      const resolved = unResolve(mySignal, {
+        // TODO @Shinigami92 2025-06-23: looks like we need to allow to pass `null`, because `undefined` will fallback to default value
```

Comment:
Is that possible to be done in this PR, I thought it would be easy enough? The castings seem quite strange to me.